### PR TITLE
(DOCSP-26666) Changeable Deployment Models

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-26666) Changeable Deployment Models

### Staged Changes

- [Change Deployment Models](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/changeable-deployment-models/apps/change-deployment-models/)
- [Admin API > (GET|PUT) /groups/{groupId}/apps/{appId}/deployment_migration](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/changeable-deployment-models/admin/api/v3/#tag/deploy/operation/adminGetDeploymentMigration)

### Reminder Checklist

You might need to also update some corresponding pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-realm update(s), if any
- [x] Checked/updated Admin API
  - included in this PR
- [x] Checked/updated CLI reference
  - not supported via CLI at this time

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/apps/change-deployment-models.txt [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+You can change an existing App's deployment configuration to deploy it
+in a new model, region, and/or cloud provider. For a list of all
+available options, see :ref:`Deployment Models & Regions
+<deployment-model>`.
+
+To move an existing App, you begin a  **Deployment Migration** that
+automatically moves your App's data and configuration to the new
+configuration. The migration process is designed to be as seamless as
+possible but necessarily requires up to 30 minutes of downtime. You can
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:21]:
```suggestion
possible but requires up to 30 minutes of downtime. You can
```

File: source/apps/change-deployment-models.txt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+2. ``"downtime"``: the App is unavailable while the migration is in progress
+
+3. ``"enabling_event_subscriptions"``: the App's Triggers and Device Sync translators are being enabled
+
+4. ``"cleanup"``: deployment artifacts are being cleaned up
+
+5. The migration is complete and can be in one of two states:
+   - ``"successful"``: the migration completed successfully
+   - ``"failed"``: the migration failed
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:195]:
```suggestion
#. ``"started"``: The migration has been started.
#. ``"downtime"``: The App is unavailable while the migration is in progress.
#. ``"enabling_event_subscriptions"``: The App's Triggers and Device Sync translators are being enabled.
#. ``"cleanup"``: Deployment artifacts are being cleaned up.
#. The migration is complete and can be in one of two states:
   - ``"successful"``: The migration completed successfully.
   - ``"failed"``: The migration failed.
```

File: source/apps/change-deployment-models.txt [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+The migration process migrates the following components of your App:
+
+- Configuration Files
+- Triggers
+- Device Sync Translators
+- API Services
+
+The migration process does not migrate the following data:
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:204]:
```suggestion
The migration process **does not** migrate the following data:
```

nit: emph

File: source/apps/change-deployment-models.txt [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+5. The migration is complete and can be in one of two states:
+   - ``"successful"``: the migration completed successfully
+   - ``"failed"``: the migration failed
+
+The migration process migrates the following components of your App:
+
+- Configuration Files
+- Triggers
+- Device Sync Translators
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:201]: what's a device sync translator?

File: source/includes/prereqs/atlas-account.rst [hunk 5]
```diff
@@ -0,0 +1,3 @@
+A MongoDB Atlas account with Project Owner permissions. To learn how to
+sign up for a free account, see :atlas:`Get Started with Atlas
+</getting-started>`.
```
Comment (mongodben) [inline source/includes/prereqs/atlas-account.rst:3]: this repo has intersphinx w atlas docs. can you ref?

File: source/includes/prereqs/atlas-admin-api-key.rst [hunk 6]
```diff
@@ -0,0 +1,4 @@
+A MongoDB Atlas Admin API :atlas:`public/private key pair
+</configure-api-access/#std-label-atlas-prog-api-key>`. The API key must
+have :atlas:`Project Owner </reference/user-roles/#project-roles>`
```
Comment (mongodben) [inline source/includes/prereqs/atlas-admin-api-key.rst:3]: this repo has intersphinx w atlas docs. can you ref?

File: source/openapi-admin-v3.yaml [hunk 7]
```diff
@@ -4315,6 +4315,49 @@ paths:
...
+              schema:
+                "$ref": "#/components/schemas/DeploymentMigration"
+        "404":
+          description: App not found
+    put:
+      tags:
+        - deploy
+      operationId: adminCreateDeploymentMigration
+      summary: Create a Deployment Migration
+      description: Start a new deployment migration
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml:4342]:
```suggestion
      description: Start a new deployment migration.
```

File: source/apps/change-deployment-models.txt [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+.. contents:: On this page
+   :depth: 2
+
+Overview
+--------
+
+You can change an existing App's deployment configuration to deploy it
+in a new model, region, and/or cloud provider. For a list of all
+available options, see :ref:`Deployment Models & Regions
+<deployment-model>`.
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:16]: can you add a short paragraph or list here about why one might want to change deployment model?

File: source/apps/change-deployment-models.txt [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+========================
+
+.. contents:: On this page
+   :depth: 2
+
+Overview
+--------
+
+You can change an existing App's deployment configuration to deploy it
+in a new model, region, and/or cloud provider. For a list of all
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:14]: > in a new model

do you mean local vs global by this? if so, could you explicitly specify?

File: source/apps/change-deployment-models.txt [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+~~~~~~~~~
+
+.. tabs-realm-admin-interfaces::
+
+   .. tab::
+      :tabid: ui
+
+      .. procedure::
+
+         .. step:: Navigate to the Deployment screen
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:68]: do we usually capitalize step titles?

**Review by mongodben — COMMENTED** [review]
looks good! just a few small things throughout

File: source/apps/change-deployment-models.txt (hunk 10 above)
Comment (mongodben) [reply source/apps/change-deployment-models.txt:84]: i mean like title case for the whole step title, so here it'd be:

```rst
.. step:: Navigate to the Deployment Screen
```

for the next:

```rst
.. step:: Choose a Target Deployment Model and Region
```

File: source/apps/change-deployment-models.txt [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+   - ``"successful"``: the migration completed successfully
+
+   - ``"failed"``: the migration failed
+
+The migration process migrates the following components of your App:
+
+- User accounts
+- Configuration files
+- Triggers
+- API Services
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:221]: > API Services

i see you did title case here..is 'API Services' now an Official Name that We Call These Things?

File: source/apps/change-deployment-models.txt [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+
+The migration process **does not** migrate the following data:
+
+- Application logs
+- Deployment drafts
+- Data stored in a MongoDB Atlas cluster
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:227]: Might be worth adding in a sentence after this that any data written by the App stays in Atlas.

Comment (mongodben) [reply source/apps/change-deployment-models.txt:227]: so readers don't think that the data is deleted from Atlas.

File: source/apps/change-deployment-models.txt [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+#. The migration is complete and can be in one of two states:
+
+   - ``"successful"``: the migration completed successfully
+
+   - ``"failed"``: the migration failed
+
+The migration process migrates the following components of your App:
+
+- User accounts
+- Configuration files
```
Comment (mongodben) [inline source/apps/change-deployment-models.txt:219]: maybe just 'App configuration'?

**Review by mongodben — APPROVED** [review]
lgtm, nice job. couple small non-blocking things that i noticed on this review, and 1 question about 'API Services' which i'm going to just ask in Slack

File: source/apps/change-deployment-models.txt (hunk 11 above)
Comment (nlarew) [reply source/apps/change-deployment-models.txt:221]: Yeah that's the plan - it's not in the product anywhere but separate API services was confusing for folks

**Pull request merged**